packages/edit-site/src/components/global-styles/typography-panel.js

```diff
@@ -1,419 +1,43 @@
-	const resetStyle = () => setStyle( undefined );
-	return [ style, setStyle, hasStyle, resetStyle ];
-}
-
-function useFontSizeWithReset( path, blockName ) {
-	const [ fontSize, setStyleCallback ] = useGlobalStyle( path, blockName );
-	const [ userStyle ] = useGlobalStyle( path, blockName, 'user' );
-	const hasFontSize = () => !! userStyle;
-	const resetFontSize = () => setStyleCallback( undefined );
-	const setFontSize = ( newValue, metadata ) => {
-		if ( !! metadata?.slug ) {
-			newValue = `var:preset|font-size|${ metadata?.slug }`;
-		}
-		setStyleCallback( newValue );
-	};
-
-	return {
-		fontSize,
-		setFontSize,
-		hasFontSize,
-		resetFontSize,
-	};
-}
-
-function useFontAppearance( prefix, name ) {
-	const [ fontStyle, setFontStyle ] = useGlobalStyle(
-		prefix + 'typography.fontStyle',
-		name
-	);
-	const [ userFontStyle ] = useGlobalStyle(
-		prefix + 'typography.fontStyle',
-		name,
-		'user'
-	);
-	const [ fontWeight, setFontWeight ] = useGlobalStyle(
-		prefix + 'typography.fontWeight',
-		name
-	);
-	const [ userFontWeight ] = useGlobalStyle(
-		prefix + 'typography.fontWeight',
-		name,
-		'user'
-	);
-	const hasFontAppearance = () => !! userFontStyle || !! userFontWeight;
-	const resetFontAppearance = () => {
-		setFontStyle( undefined );
-		setFontWeight( undefined );
-	};
-	return {
-		fontStyle,
-		setFontStyle,
-		fontWeight,
-		setFontWeight,
-		hasFontAppearance,
-		resetFontAppearance,
-	};
-}
+const { useGlobalStyle, TypographyPanel: StylesTypographyPanel } = unlock(
+	blockEditorExperiments
+);
 
 export default function TypographyPanel( {
 	name,
 	element,
 	headingLevel,
 	variationPath = '',
 } ) {
-	const supports = getSupportedGlobalStylesPanels( name );
 	let prefix = '';
 	if ( element === 'heading' ) {
 		prefix = `elements.${ headingLevel }.`;
 	} else if ( element && element !== 'text' ) {
 		prefix = `elements.${ element }.`;
 	}
-	const [ fontSizes ] = useGlobalSetting( 'typography.fontSizes', name );
-
-	const disableCustomFontSizes = ! useGlobalSetting(
-		'typography.customFontSize',
-		name
-	)[ 0 ];
-	const [ fontFamilies ] = useGlobalSetting(
-		'typography.fontFamilies',
-		name
-	);
-	const hasFontStyles =
-		useGlobalSetting( 'typography.fontStyle', name )[ 0 ] &&
-		supports.includes( 'fontStyle' );
-	const hasFontWeights =
-		useGlobalSetting( 'typography.fontWeight', name )[ 0 ] &&
-		supports.includes( 'fontWeight' );
-	const hasFontFamilyEnabled = useHasFontFamilyControl( name );
-	const hasLineHeightEnabled = useHasLineHeightControl( name );
-	const hasAppearanceControl = useHasAppearanceControl( name );
-	const appearanceControlLabel = useAppearanceControlLabel( name );
-	const hasLetterSpacingControl = useHasLetterSpacingControl( name, element );
-	const hasTextTransformControl = useHasTextTransformControl( name, element );
-	const hasTextDecorationControl = useHasTextDecorationControl(
+	const [ style, setStyle ] = useGlobalStyle( variationPath + prefix, name );
```

Comment:
It just occurred to me reading this how strange it is that `setStyle` always sets a `'user'` style even when `'base'` is provided.

```js
const [ , setStyle ] = useGlobalStyle( path, name, 'base' );
```

Reply:
yeah, we could potentially split these hooks into to hooks: one to retrieve the value and one to update it, the latter doesn't need the source.